src/cryptoadvance/specter/user.py

```diff
@@ -86,14 +88,14 @@ def __init__(
         self.wallet_manager = None
         self.device_manager = None
         self.manager = None
-        self._services = services
+        self._services = services if services else User.default_services
```

Comment:
I don't like that the `default_services` are specified in the User class. This exposes way too much details to the the User class which does not belong there.
In general, Businessclasses which represent Businessobjects should also not gather informations by somehow asking Managers. From an architecture point of view, i'd remove the `self.wallet_manager` and `self.device_manager` and rather inject the details at instatiation-time (or afterwards via setters). Having quite passive Businesspbjects is usually a good idea. Anyway, that was just a sidenote.
So i'd suggest to inject the `services` with no alternative here. Instantiation of User-Classes are hopefully mostly at the `UserManager` and there, this should be asked at the `ServiceManager`.
It's questionable why it should even be hardcoded there: IMHO the `ServiceManager` should iterate all the extensions and maybe ask them for a new property maybe called `opt` which could be `opt-in`, `opt-out` or `mandatory`.
Whereas `opt-out` combined with `mandatory` would be the list which would end up here.
The difference between 'opt-out' and 'mandatory' would be, that the mandatory extension doesn't give you an option in it's settings, to deactivate. Later, we could put the deactivation procedure additionally on a technical page for each service which is entirely managed by "Core". Whether the extension is respecting it's deactivation is still something each extension need to deal with. Limiting that responsibility the more we move forward makes sense as well (e.g. not calling callbacks for deactivated extensions but there is more to that).
Does that make sense? I think this would also address your concerns regarding the semantics of "activation" we discussed recently.

Reply:
> `opt-in`, `opt-out` or `mandatory`.

I like that. I will do it that way.

Reply:
I addressed this in https://github.com/cryptoadvance/specter-desktop/pull/1766/commits/ecf5da0090adda2c447c78e100e119576b9b3092 . However the migration is done very crudely. Do you know a more elegant way to migrate the user configs?

Reply:
I can't see the crudeness in this. Migration is almost never something elegant but in that case it even ist. Using the migration-classes is elegant in itself but i might be biased with this.
But maybe i miss the point?!